shotover-proxy/src/transforms/redis/sink_cluster.rs

```diff
@@ -547,27 +565,33 @@ enum ChannelResult {
 
 impl RoutingInfo {
     #[inline(always)]
-    pub fn for_command_frame(args: &[RedisFrame]) -> Result<Option<RoutingInfo>> {
+    pub fn for_command_frame(args: &[RedisFrame]) -> Result<RoutingInfo> {
         let command_name = match args.get(0) {
             Some(RedisFrame::BulkString(command_name)) => command_name.to_ascii_uppercase(),
             _ => bail!("syntax error: bad command name"),
         };
 
         Ok(match command_name.as_slice() {
-            b"FLUSHALL" | b"FLUSHDB" => Some(RoutingInfo::AllMasters),
+            b"FLUSHALL" | b"FLUSHDB" => RoutingInfo::AllMasters(ResponseJoin::First),
             b"ECHO" | b"CONFIG" | b"CLIENT" | b"SLOWLOG" | b"DBSIZE" | b"LASTSAVE" | b"PING"
-            | b"INFO" | b"BGREWRITEAOF" | b"BGSAVE" | b"SAVE" | b"TIME" | b"KEYS" | b"ACL" => {
-                Some(RoutingInfo::AllNodes)
+            | b"INFO" | b"BGREWRITEAOF" | b"BGSAVE" | b"SAVE" | b"TIME" | b"ACL" => {
+                RoutingInfo::AllNodes
             }
+
+            b"KEYS" => RoutingInfo::AllMasters(ResponseJoin::ArrayJoin),
             b"SCRIPT" => match args.get(1) {
-                Some(RedisFrame::BulkString(a)) if a.to_ascii_uppercase() == b"KILL" => None,
-                _ => Some(RoutingInfo::AllMasters),
+                Some(RedisFrame::BulkString(a)) if a.to_ascii_uppercase() == b"KILL" => {
+                    RoutingInfo::Unsupported
+                }
+                _ => RoutingInfo::AllMasters(ResponseJoin::First),
             },
-            b"SCAN" | b"SHUTDOWN" | b"SLAVEOF" | b"REPLICAOF" | b"MOVE" | b"BITOP" => None,
+            b"SCAN" | b"SHUTDOWN" | b"SLAVEOF" | b"REPLICAOF" | b"MOVE" | b"BITOP" => {
+                RoutingInfo::Unsupported
```

Comment:
What is the practical implication here, if these commands are submitted to shotover, they are dropped completely?

Reply:
No messages are sent to the sink redis instances and an error is returned to the client.